Made KeyVal public to support using iterators outside of the package

RobertAlblas opened the pull request:
KeyVal is used in src/test/java/org/lmdbjava/TutorialTest.java to demonstrate iterating over the entire collection. This does not work for client code, since KeyVal is currently package protected.

benalexau commented:
`TutorialTest` demonstrates the use of `CursorIterator`, the latter of which exposes (via its `iterable()` method) the `public` `org.lmdbjava.CursorIterator.KeyVal`. The `CursorIterator.KeyVal` is not the package-protected `org.lmdbjava.KeyVal`.

To confirm `TutorialTest` operates outside the `org.lmdbjava` package I moved it to `com`, fixed the imports, and re-ran it successfully:

```
Running com.TutorialTest
Tests run: 6, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.527 s - in com.TutorialTest
```

As such I don't believe we need to change the visibility of `org.lmdbjava.KeyVal`.

RobertAlblas commented:
Looks good, thank you for the clarification.

On Mon, 11 Jun 2018, 22:11 Ben Alex, <notifications@github.com> wrote:

> TutorialTest demonstrates the use of CursorIterator, the latter of which
> exposes (via its iterable() method) the public
> org.lmdbjava.CursorIterator.KeyVal. The CursorIterator.KeyVal is not the
> package-protected org.lmdbjava.KeyVal.
>
> To confirm TutorialTest operates outside the org.lmdbjava package I moved
> it to com, fixed the imports, and re-ran it successfully:
>
> Running com.TutorialTest
> Tests run: 6, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.527 s - in com.TutorialTest
>
> As such I don't believe we need to change the visibility of
> org.lmdbjava.KeyVal.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/lmdbjava/lmdbjava/pull/93#issuecomment-396370836>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AFo1WvPWPm7WK0_Jw7lmP-0fkJqa5h7Lks5t7s7igaJpZM4Ua1M5>
> .
>
